coral/src/app/features/topics/acl-request/TopicAclRequest.tsx

```diff
@@ -0,0 +1,203 @@
+  const navigate = useNavigate();
+  const [topicType, setTopicType] = useState("Producer");
+
+  useEffect(() => {
+    if (window.msw !== undefined) {
+      mockGetEnvironments({
+        mswInstance: window.msw,
+        response: { data: mockedData },
+      });
+      mockGetTopicNames({
+        mswInstance: window.msw,
+        response: mockedResponseTopicNames,
+      });
+      mockGetTopicTeam({
+        mswInstance: window.msw,
+        response: mockedResponseTopicTeamLiteral,
+        topicName,
+      });
+      mockGetClusterInfoFromEnv({
+        mswInstance: window.msw,
+        response: getMockedResponseGetClusterInfoFromEnv(true),
+      });
+    }
+  }, []);
+
+  const topicProducerForm = useForm<TopicProducerFormSchema>({
+    schema: topicProducerFormSchema,
+    defaultValues: {
+      topicname: topicName,
+      environment: "placeholder",
+      topictype: "Producer",
+    },
+  });
+
+  const topicConsumerForm = useForm<TopicConsumerFormSchema>({
+    schema: topicConsumerFormSchema,
+    defaultValues: {
+      aclPatternType: "LITERAL",
+      topicname: topicName,
+      environment: "placeholder",
+      topictype: "Consumer",
+    },
+  });
+
+  const { data: topicNames } = useQuery<TopicNames, Error>(["topic-names"], {
+    ...topicNamesQuery(),
+    onSettled: (data) => {
+      if (data?.includes(topicName)) {
+        return;
+      }
+      // Navigate back to Topics when topicName does not exist in the topics list
+      navigate("/topics");
+    },
+    enabled: topicName !== "",
+  });
+
+  const { data: environments } = useQuery<Environment[], Error>(
+    ["topic-environments"],
+    {
+      queryFn: getEnvironments,
+    }
+  );
+  const { data: topicTeam } = useQuery<TopicTeam, Error>(
+    ["topic-team", topicName],
+    topicTeamQuery({ topicName })
+  );
+
+  const selectedEnvironment =
+    topicType === "Producer"
+      ? topicProducerForm.watch("environment")
+      : topicConsumerForm.watch("environment");
+  const selectedEnvironmentType =
+    environments?.find((env) => env.id === selectedEnvironment)?.type || "";
+  const { data: clusterInfo } = useQuery<ClusterInfo, Error>(
+    ["cluster-info", selectedEnvironment],
+    {
+      queryFn: () =>
+        getClusterInfo({
+          envSelected: selectedEnvironment,
+          envType: selectedEnvironmentType,
+        }),
+
+      keepPreviousData: false,
+      enabled:
+        selectedEnvironment !== "placeholder" && environments !== undefined,
+      onSettled: (data) => {
+        const isAivenCluster = data?.aivenCluster === "true";
+        // Enable the only possible option when the environment chosen is Aiven Kafka flavor
+        if (isAivenCluster) {
+          return topicType === "Producer"
+            ? topicProducerForm.setValue("aclIpPrincipleType", "PRINCIPAL")
+            : topicConsumerForm.setValue("aclIpPrincipleType", "PRINCIPAL");
+        }
+      },
+    }
+  );
+
+  if (
+    topicNames === undefined ||
```

Comment:
For sure, I planned to do this in the subsequent PR. Howwver, an interesting point: even if `isLoading` is true, the `data` type signature returned from a `useQuery` will be `Type | undefined`. So the check seems to always have to be `isLoading && data !== undefined`. Is there a better way to avoid this double check I did not think about?

Reply:
Yeah! Lets not get stuck with this detail. It will be more peasant for everyone to change in a followup PR :)